src/main/webapp/js/components/app.jsx

```diff
@@ -61,10 +61,12 @@ class App extends Component {
     return (
       <Router>
         <Switch>
+          {this.props.authenticated
+            ? <RoutePrivate path={this.props.history.path} component={AppContent} />
```

Comment:
I don't think changing the path is the solution here, you should look into the initial value of the authenticated property.
I think it's the issue with the redux offline, when refreshing or opening a new tab the redux state has initial value, the correct values are later loaded from the redux offline, but before this happens the Redirect in the PrivateRoute is rendered.